src/core/core.interaction.js

```diff
@@ -279,10 +280,19 @@ module.exports = {
 			var intersectsItem = false;
 
 			parseVisibleItems(chart, function(element) {
-				if (element.inXRange(position.x)) {
-					items.push(element);
+				if (element.datasetlength < 40) {
+					if (element.inXRange(position.x)) {
+						items.push(element);
+					}
+				} else if (element.datasetlength >= 40) {
```

Comment:
you could turn this `else if` into simply an `else`

however, i'm not sure why the check for whether there are more or less than 40 elements is being added. Why not always to do the cleanup of extra elements?

Reply:
You're right the `else if` check is kind of redundant. After some extensive testing, I found the issue occurs after there are 40 datapoints on the chart. From my understanding, the reason for `inXRange` is to allow the user to not have to be precisely on the 'x' coordinate to have the datapoint highlighted by the tooltip and rather have the user be in a certain range of the tooltip making it easier to highlight. However, the main problem here is when we have say 300+ datapoints there begins to be too many data points that fall within that same range. Overall the having a separate function `inXRangeSmall` narrows the range in which a datapoint will be returned if we have an extensive amount of datapoints (40+). Also I can add the cleanup to both just in case. Thanks for the review!

Reply:
When I convert the `else if` to `else` I get the "no-lonely-if" lint error. 

![chartjsif](https://user-images.githubusercontent.com/27080760/38462068-d2d5ce98-3aad-11e8-94af-d4e93c2fa071.JPG)

Reply:
It wants you to combine the else with the next if: `} else iif (element.inXRangeSmall(position.x)) {`

I still am not understanding though why we do not always call `inXRangeSmall` and remove the check for whether there are 40 datapoints